module/zfs/zfs_ioctl.c

```diff
@@ -2415,6 +2450,11 @@ zfs_prop_set_special(const char *dsname, zprop_source_t source,
 		if (err == 0)
 			err = -1;
 		break;
+	case ZFS_PROP_KEYLOCATION:
+		err = dsl_crypto_can_set_keylocation(dsname, strval);
+		if (err == 0)
+			err = -1;
```

Comment:
Yeah I saw that but I know I've puzzled over things like this before because I only looked at the code I thought was relevant to the code path that was being taken (i.e. not the other properties - and I could imagine in the future code being added between this and the earlier comment such that they are not so close together.)